xurls.go

```diff
@@ -77,7 +77,12 @@ func relaxedExp() string {
 	site := domain + `(?i)(` + punycode + `|` + knownTLDs + `)(?-i)`
 	hostName := `(` + site + `|` + ipAddr + `)`
 	webURL := hostName + port + `(/|/` + pathCont + `?|\b|(?m)$)`
-	return strictExp() + `|` + webURL
+
+	nonASCIISite :=  domain + `(?i)(` + punycode + `|` + anyOf(NonASCIITLDs...) + `)(?-i)`
+	nonASCIIHostName := `(` + nonASCIISite + `|` + ipAddr + `)`
+	nonASCIIWebURL := nonASCIIHostName + port + `(/|/` + pathCont + `?|\B|(?m)$)`
```

Comment:
I don't think the `\B` is useful here.

Also, you duplicate quite a lot of the logic, so I think this makes the code less maintainable.